src/responder/common/cache_req/cache_req.c

```diff
@@ -1275,6 +1275,15 @@ static void cache_req_input_parsed(struct tevent_req *subreq)
     state = tevent_req_data(req, struct cache_req_state);
 
     ret = sss_parse_inp_recv(subreq, state, &name, &domain);
+
+    if (state->domain_name && domain && strcmp(state->domain_name, domain)){
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "Mismatch between input domain name [%s] and parsed domain name [%s]\n",
+              state->domain_name, domain);
+        tevent_req_error(req, EIO); //TODO: change error code?
```

Comment:
@pbrezina I've just noticed that I forgot to ask about this `TODO`. Do you think it's fine to return `EIO`? From my point of view this is the error code that suits best, but I'm open to changing it.

Reply:
EIO or perhaps EINVAL. ERR_INPUT_PARSE,

Reply:
I like `ERR_INPUT_PARSE` the most. Thank you.